docker/jupyter/custom/JupyterLockingExtension.js

```diff
@@ -0,0 +1,325 @@
+const dialog = require('base/js/dialog')
+const utils = require("base/js/utils")
+
+define(() => {
+    var modalOpen = false
+    var meta = {}
+
+    const syncIssueButtons = {
+        'Make a Copy': {
+            click: () => saveAs()
+        },
+        'Reload the workspace version and discard your changes': {
+            click: () => updateLocalCopyWithRemote()
+        }
+    }
+
+    const lockIssueButtons = {
+        'Run in Playground Mode': {
+            click: () => openPlaygroundMode(),
+            'class': 'btn-primary'
+        },
+        'Make a Copy': {
+            click: () => saveAs()
+        }
+    }
+
+    const noRemoteFileButtons = {
+        'Continue working': {
+            click: () => {}
+        }
+    }
+
+    const modeBannerId = "notification_mode"
+    const lockConflictTitle = "File is in use"
+    const syncIssueTitle = "File versions out of sync"
+    const syncIssueBody = "Your version of this file does not match the version in the workspace. What would you like to do?"
+    const syncIssueFatalBody = "This file was either deleted or never was stored with us."
+
+    //TODO URL resolution
+    const leoUrl = 'http://localhost:8080'
```

Comment:
Because this is client-side, localhost won't work.

The browser running the notebook will set an auth cookie so we will be able to use relative paths here, which will call out to the Leo proxy.  `welderUrl` is `proxy/<billing project>/<cluster name>/welder`

Reply:
Do you have any idea how to access this cookie or how I can test this? This has been something I haven't been sure how to test